### ISSUE TITLE
Tempus: Add Stepper Capability to Initialize Within Individual Set Calls

### ISSUE THREAD

**Issue by ccober6** [trilinos/Trilinos#4867]
One path for Stepper construction is to start with a default constructor and then use set* methods to specify parameters, and features of the Stepper for the current problem.  After this the Stepper needs to call initialize() to finish the changes.

Here we would like to have each set function optional call initialize() internally, so that it is less likely to forget the initialize() call before starting the time integration.

@trilinos/tempus 

## Expectations
Have the ability to call initialize within each set function, and multiple calls initialize is possible.

## Definition of Done
- [ ] Add optional calls to initialize from within set functions.
- [ ] Pass all tests.
- [ ] Add test(s) illustrating functionality.

**Comment by rppawlo** [comment]
@ccober6 - does initialize do anything that would be costly at scale? Another option would be to save an internal bool "is_initialized" that can throw at runtime if the stepper is used without initialization being called. Each set method can set teh stepper back to an "uninitialized" state with the bool. My worry is that I can see users just always calling initialize to be safe and  then later on trying to figure out why the setup phase is so inefficient.

**Comment by ccober6** [comment]
@rppawlo  Yeah, I hear you.  So there are two things going on here:

1. Initializing within set calls (A) versus only allowing it explicitly after set calls (B)
    * Convenience and assurance with (A)
    * Controlling/highlighting use with (B)
    * User flexibility at the risk of performance.
2. Ensure stepper is initialized with "is_initialized" before use.
    * I have stayed away from this as the only place to check it is that the beginning of takeStep().
       - A little ugly.  Rythmos put a lot of setup and initialization stuff in its takeStep(), which really cluttered the code (and I believe slowed it down), so I have been avoiding it.
    * With that said, I have not found another means to ensure the stepper "is_initialized", so I am seriously contemplating the "is_initialized" approach.
    * This is a code "safety" issue.

So right now, I am looking at the following set functions:
 * setTableau()
 * setTableaus()
 * setParameterList()
 * setModel()
 * createSubSteppers()
 * setSolver()
 * setPredictor()
 * setStartUpStepper()
 * setObserver()
None of these are doing anything computational expense, so I am not too worried about it.  But we both know that changes and things get abused.

I will likely go with you suggestion.

**Comment by ccober6** [comment]
Closing in favor of #5908.
